From 11a1879976aff596ae5fcc91616a56e214044748 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Thu, 24 Aug 2017 18:11:28 +0800
Subject: [PATCH 86/87] ARM: sun8i: r40: enable Ethernet on Banana Pi M2 Ultra

Banana Pi M2 Ultra features an Ethernet port, connected to the R40 SoC's
GMAC via a RTL8211E PHY.

Enable it in the device tree.

ALDO2 regulator drives the RGMII IO on the SoC, and should be enabled.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 arch/arm/boot/dts/sun8i-r40-bananapi-m2-ultra.dts | 27 +++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/arm/boot/dts/sun8i-r40-bananapi-m2-ultra.dts b/arch/arm/boot/dts/sun8i-r40-bananapi-m2-ultra.dts
index cc91d02a3b31..d6b379171801 100644
--- a/arch/arm/boot/dts/sun8i-r40-bananapi-m2-ultra.dts
+++ b/arch/arm/boot/dts/sun8i-r40-bananapi-m2-ultra.dts
@@ -108,6 +108,15 @@
 	status = "okay";
 };
 
+&gmac {
+	pinctrl-names = "default";
+	pinctrl-0 = <&gmac_rgmii_pins>;
+	phy-supply = <&reg_dc1sw>;
+	phy-handle = <&ext_rgmii_phy>;
+	phy-mode = "rgmii";
+	status = "okay";
+};
+
 &i2c0 {
 	status = "okay";
 
@@ -121,6 +130,13 @@
 
 #include "axp22x.dtsi"
 
+&reg_aldo2 {
+	regulator-always-on;
+	regulator-min-microvolt = <2500000>;
+	regulator-max-microvolt = <2500000>;
+	regulator-name = "vcc-pa";
+};
+
 &reg_aldo3 {
 	regulator-always-on;
 	regulator-min-microvolt = <2700000>;
@@ -128,6 +144,10 @@
 	regulator-name = "avcc";
 };
 
+&reg_dc1sw {
+	regulator-name = "vcc-phy";
+};
+
 &reg_dcdc1 {
 	regulator-always-on;
 	regulator-min-microvolt = <3000000>;
@@ -180,6 +200,13 @@
 	regulator-name = "vdd-1v2-sata";
 };
 
+&mdio {
+	ext_rgmii_phy: ethernet-phy@1 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <1>;
+	};
+};
+
 &mmc0 {
 	vmmc-supply = <&reg_dcdc1>;
 	bus-width = <4>;
-- 
2.13.5

